services/logs/filter.go

```diff
@@ -86,16 +84,26 @@ func (r *RangeFilter) Match() ([]*gethTypes.Log, error) {
 		return nil, err
 	}
 
-	logs := make([]*gethTypes.Log, 0)
+	var bloomHeightMatches []uint64
+	var logs []*gethTypes.Log
+
+	// first filter all the logs based on whether a bloom matches,
+	// if bloom matches we fetch only that height later and do exact match
 	for _, bloomHeight := range bloomsHeight {
 		for _, bloom := range bloomHeight.Blooms {
-			if !bloomMatch(*bloom, r.criteria) {
```

Comment:
this was broken, probably got broken after changes, because it did nothing, so it didn't optimize next step.